### ISSUE TITLE
Option to show a sorted list of patches in their respective categories

### ISSUE THREAD

**Issue by mkruselj** [samlletas/syxg50-control-panel#16]
So now that we're not showing MSB/LSB/PC numbers anywhere, maybe we can sort the patches so that numbered variations are all grouped together? Or just outright no option and we just sort things out somehow?

**Comment by samlletas** [comment]
Hmmm looking at XG works it seems voice variations are nested, for example:

- Grand Piano
  - Grand Piano KSP
  - Mellow Grand Piano
  - Piano Strings
  - Dream
- Bright Piano
  - Bright Piano KSP
 - etc...

The control panel follows the same order (but without nesting) so we could probably group them together without changing the order of the voice list.

**Comment by mkruselj** [comment]
I'm not a fan of that nesting, just makes things slower...

BTW I was not talking about dropdown menus in the Overview page, but about the Voice List instead. These situations for example:

![image](https://github.com/user-attachments/assets/a4d9f3bd-2d2b-4d31-a189-318a327bd55c)
![image](https://github.com/user-attachments/assets/87280ffe-b247-463a-9a0b-1c366c87016a)
![image](https://github.com/user-attachments/assets/6707be06-5943-4053-9003-eec22411b503)


etc.

**Comment by samlletas** [comment]
Ah I see what you meant now, I think outright changing the order in the `...data.jsfx-inc` file is something I would avoid since that's a breaking change (the selected voice index is serialized), specially after v3 is already out.

We could add a sorting option instead (perhaps by name?) and adjust the selected voice index when changing sorting modes (e.g comparing program and bank numbers since those don't change).

**Comment by mkruselj** [comment]
Yes, there should be a remapping array for this in order to retain the streaming.
